packages/widgets/src/menubar.ts

```diff
@@ -737,11 +749,11 @@ export class MenuBar extends Widget {
   }
 
   /**
-   * Handle the `'mouseleave'` event for the menu bar.
+   * Handle the `'focusout'` event for the menu bar.
    */
-  private _evtMouseLeave(event: MouseEvent): void {
-    // Reset the active index if there is no open menu.
-    if (!this._childMenu) {
+  private _evtFocusOut(event: FocusEvent): void {
```

Comment:
The MenuBar class was originally written to handle mouse and keyboard interaction; however, it really wasn't concerned with browser focus. For example, in the original version, you could use the left and right arrow keys to "move" from one item to another in the menu bar, but browser focus did not move when you pressed those keys. 

There was no way to even put browser focus on individual item nodes in the menu bar until https://github.com/jupyterlab/lumino/pull/187. As a result, it made sense to set the active index to -1 when the mouse left the menu bar. But now that the active index is closely tied to focus, to be consistent we shouldn't reset the active index just because the mouse leaves the menu bar. 

However, we can safely reset the active index if the menu bar loses focus.